We shouldn't be suppressing this. This might be a bug in the latest ILLink analyzer (quite possibly from https://github.com/dotnet/runtime/pull/92724 ?).

@sbomer - can you take a look? Just pull this PR and `.\eng\build.cmd` from the root of the repo. You should see:

```
C:\git\aspnetcore\src\JSInterop\Microsoft.JSInterop\src\Infrastructure\DotNetDispatcher.cs(391,67): error IL2111: Method 'ScanTypeForCallableMethods(
Type)' with parameters or return value with `DynamicallyAccessedMembersAttribute` is accessed via reflection. Trimmer can't guarantee availability of
 the requirements of the method. [C:\git\aspnetcore\src\JSInterop\Microsoft.JSInterop\src\Microsoft.JSInterop.csproj]
```

Note that this method is a local / nested method, so I'm not sure how it could be "accessed via reflection".

https://github.com/dotnet/aspnetcore/blob/82ff9246592477abf4ff57ac905a368cbd99ff20/src/JSInterop/Microsoft.JSInterop/src/Infrastructure/DotNetDispatcher.cs#L388-L405

BTW - @amcasey, there are other build errors in this PR:

```
C:\git\aspnetcore\src\SignalR\clients\csharp\Client.Core\src\HubConnection.cs(866,47): error CA2007: Consider calling ConfigureAwait on the awaited t
ask (https://learn.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/ca2007) [C:\git\aspnetcore\src\SignalR\clients\csharp\Client.Core\sr
c\Microsoft.AspNetCore.SignalR.Client.Core.csproj::TargetFramework=net9.0]
C:\git\aspnetcore\src\SignalR\clients\csharp\Client.Core\src\HubConnection.cs(866,47): error CA2007: Consider calling ConfigureAwait on the awaited t
ask (https://learn.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/ca2007) [C:\git\aspnetcore\src\SignalR\clients\csharp\Client.Core\sr
c\Microsoft.AspNetCore.SignalR.Client.Core.csproj::TargetFramework=netstandard2.0]
C:\git\aspnetcore\src\SignalR\clients\csharp\Client.Core\src\HubConnection.cs(866,47): error CA2007: Consider calling ConfigureAwait on the awaited t
ask (https://learn.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/ca2007) [C:\git\aspnetcore\src\SignalR\clients\csharp\Client.Core\sr
c\Microsoft.AspNetCore.SignalR.Client.Core.csproj::TargetFramework=netstandard2.1]
C:\git\aspnetcore\src\SignalR\clients\csharp\Client.Core\src\HubConnection.cs(866,47): error CA2007: Consider calling ConfigureAwait on the awaited t
ask (https://learn.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/ca2007) [C:\git\aspnetcore\src\SignalR\clients\csharp\Client.Core\sr
c\Microsoft.AspNetCore.SignalR.Client.Core.csproj::TargetFramework=net462]
```

--- a/src/JSInterop/Microsoft.JSInterop/src/Infrastructure/DotNetDispatcher.cs
+++ b/src/JSInterop/Microsoft.JSInterop/src/Infrastructure/DotNetDispatcher.cs
@@ -388,7 +388,12 @@ public static class DotNetDispatcher
     private static (MethodInfo methodInfo, Type[] parameterTypes) GetCachedMethodInfo(IDotNetObjectReference objectReference, string methodIdentifier)
     {
         var type = objectReference.Value.GetType();
+
+        // Suppressed with "pragma warning disable" instead of WarningSuppressions.xml so ILLink Roslyn Anayzer doesn't report the warning.
+#pragma warning disable IL2111 // Method with parameters or return value with `DynamicallyAccessedMembersAttribute` is accessed via reflection. Trimmer can't guarantee availability of the requirements of the method.
         var assemblyMethods = _cachedMethodsByType.GetOrAdd(type, ScanTypeForCallableMethods);
+#pragma warning restore IL2111 // Method with parameters or return value with `DynamicallyAccessedMembersAttribute` is accessed via reflection. Trimmer can't guarantee availability of the requirements of the method.
+
         if (assemblyMethods.TryGetValue(methodIdentifier, out var result))
         {
             return result;

--- a/src/JSInterop/Microsoft.JSInterop/src/Microsoft.JSInterop.WarningSuppressions.xml
+++ b/src/JSInterop/Microsoft.JSInterop/src/Microsoft.JSInterop.WarningSuppressions.xml
@@ -1,4 +1,4 @@
-﻿<?xml version="1.0" encoding="utf-8"?>
+<?xml version="1.0" encoding="utf-8"?>
 <linker>
   <assembly fullname="Microsoft.JSInterop, Version=7.0.0.0, Culture=neutral, PublicKeyToken=adb9793829ddae60">
     <attribute fullname="System.Diagnostics.CodeAnalysis.UnconditionalSuppressMessageAttribute">
@@ -66,12 +66,6 @@
       <argument>IL2091</argument>
       <property name="Scope">member</property>
       <property name="Target">M:Microsoft.JSInterop.JSRuntimeExtensions.&lt;InvokeAsync&gt;d__4`1.MoveNext</property>
-    </attribute>
-    <attribute fullname="System.Diagnostics.CodeAnalysis.UnconditionalSuppressMessageAttribute">
-      <argument>ILLink</argument>
-      <argument>IL2111</argument>
-      <property name="Scope">member</property>
-      <property name="Target">M:Microsoft.JSInterop.Infrastructure.DotNetDispatcher.GetCachedMethodInfo(Microsoft.JSInterop.Infrastructure.IDotNetObjectReference,System.String)</property>
     </attribute>
   </assembly>
 </linker>

--- a/src/JSInterop/Microsoft.JSInterop/src/Microsoft.JSInterop.WarningSuppressions.xml
+++ b/src/JSInterop/Microsoft.JSInterop/src/Microsoft.JSInterop.WarningSuppressions.xml
@@ -67,5 +67,11 @@
       <property name="Scope">member</property>
       <property name="Target">M:Microsoft.JSInterop.JSRuntimeExtensions.&lt;InvokeAsync&gt;d__4`1.MoveNext</property>
     </attribute>
+    <attribute fullname="System.Diagnostics.CodeAnalysis.UnconditionalSuppressMessageAttribute">
+      <argument>ILLink</argument>
+      <argument>IL2111</argument>
+      <property name="Scope">member</property>
+      <property name="Target">M:Microsoft.JSInterop.Infrastructure.DotNetDispatcher.GetCachedMethodInfo(Microsoft.JSInterop.Infrastructure.IDotNetObjectReference,System.String)</property>
+    </attribute>
   </assembly>
 </linker>

--- a/src/SignalR/clients/csharp/Client.Core/src/HubConnection.cs
+++ b/src/SignalR/clients/csharp/Client.Core/src/HubConnection.cs
@@ -863,7 +863,7 @@ public partial class HubConnection : IAsyncDisposable
         {
             var streamValues = AsyncEnumerableAdapters.MakeCancelableTypedAsyncEnumerable(stream, tokenSource);
 
-            await foreach (var streamValue in streamValues)
+            await foreach (var streamValue in streamValues.ConfigureAwait(false))
             {
                 await SendWithLock(connectionState, new StreamItemMessage(streamId, streamValue), tokenSource.Token).ConfigureAwait(false);
                 Log.SendingStreamItem(_logger, streamId);

